pkg/session/session.go

```diff
@@ -1838,6 +1838,7 @@ func (s *session) useCurrentSession(execOption sqlexec.ExecOption) (*session, fu
 	prevSQL := s.sessionVars.StmtCtx.OriginalSQL
 	prevStmtType := s.sessionVars.StmtCtx.StmtType
 	prevTables := s.sessionVars.StmtCtx.Tables
+	prevTypeFlags := s.sessionVars.StmtCtx.TypeFlags()
```

Comment:
Why do we need this change? Could you explain a little bit?

Reply:
The `useCurrentSession` will return a func as the `clean` func after the session executes a SQL.
We need to make sure that the CLEAN operation will not clean the flag we set.